src/meds_etl/simple_csv.py

```diff
@@ -0,0 +1,220 @@
+import argparse
+import json
+import multiprocessing
+import os
+import pickle
+import random
+import shutil
+import subprocess
+import tempfile
+
+import jsonschema
+import meds
+import polars as pl
+import pyarrow.parquet as pq
+
+import meds_etl
+
+
+def load_file(decompressed_dir, fname):
+    if fname.endswith(".gz"):
+        file = tempfile.NamedTemporaryFile(dir=decompressed_dir)
+        subprocess.run(["gunzip", "-c", fname], stdout=file)
+        return file
+    else:
+        return open(fname)
+
+
+def process_file(args):
+    event_file, num_shards, temp_dir, decompressed_dir, index = args
+    print("Working on ", event_file)
+
+    with load_file(decompressed_dir, event_file) as temp_f:
+        table = pl.read_csv_batched(temp_f.name, infer_schema_length=0, batch_size=1_000_000)
```

Comment:
Why not just https://pola-rs.github.io/polars/py-polars/html/reference/api/polars.scan_csv.html

Reply:
The issue is that  .partition_by later in the code (which is used for sharding) requires everything to be loaded into memory. So we have to explicitly batch here.

Reply:
I think we can do a scan, then collect at the end, right? Or you don't want to collect at the end for RAM usage purposes prior to the partitioned write?

Reply:
Yeah, we don't want to collect at the end due to RAM usage reasons.